src/libs/calculateAnchorPosition.ts

```diff
@@ -0,0 +1,32 @@
+/* eslint-disable no-console */
+import {ValueOf} from 'type-fest';
+import {View} from 'react-native';
+import CONST from '../CONST';
+
+type AnchorOrigin = {
+    horizontal: ValueOf<typeof CONST.MODAL.ANCHOR_ORIGIN_HORIZONTAL>;
+    vertical: ValueOf<typeof CONST.MODAL.ANCHOR_ORIGIN_VERTICAL>;
+    shiftVertical?: number;
+};
+
+type AnchorPosition = {
+    horizontal: number;
+    vertical: number;
+};
+
+/**
+ * Gets the x,y position of the passed in component for the purpose of anchoring another component to it.
+ */
+export default function calculateAnchorPosition(anchorComponent: View, anchorOrigin: AnchorOrigin): Promise<AnchorPosition> {
```

Comment:
Shouldn't `anchorOrigin` be optional as described in JSDoc comment?

```
@param {{horizontal: string, vertical: string}} anchorOriginValue - Optional parameter
```

Reply:
Yep, good catch 👌